### DISCUSSION TITLE
Curious on why `display: table` is used for scroll area content?

### DISCUSSION THREAD

**Discussion by errmayank** [melt-ui/melt-ui#1093] [Q&A]
I have a nested scroll area setup, where one scroll area is for the main page and other nested inside it for a text editor, i ran into this weird bug with inner scroll area not adjusting to the screen width, and upon further digging i found out the bug goes away when i remove the pre-applied style `display: table` for the scroll area content

This is the exact line that's causing it:
https://github.com/melt-ui/melt-ui/blob/fc29cba812b01de27da55ccd0e7957494285aec4/src/lib/builders/scroll-area/create.ts#L197

https://github.com/melt-ui/melt-ui/assets/64462998/16326a34-1d86-4975-b1a3-dbaee72e82b8

I'm not much familiar with accessibility so i'm not sure if using `display: table` has something to do with it or not 🤔

Here's the reproduction code:
https://github.com/errmayank/shadcn-svelte-scroll-bug

**Comment by errmayank** [comment]
I was able to fix the width issue by setting `display: grid` on the immediate parent of the nested scroll area, but i'm not sure what impact it might have, or is it even okay to have a custom scroll area for the whole page because of accessibility or performance reasons

**Comment by brymer-meneses** [comment]
This seems to be related to this issue https://github.com/radix-ui/primitives/issues/926 
 https://github.com/radix-ui/primitives/issues/926#issuecomment-1111961689
